gateway-service/src/main/resources/configs/common/application.conf

```diff
@@ -40,39 +40,18 @@ interaction.config = [
 
 timestamp.config = []
 
-domainobject.config = [
+entity.idcolumn.config = [
```

Comment:
@buchi-busireddy @aaron-steinfeld This is the major transformation

Reply:
So we still need the id columns? Doesn't shock me since we do in graphql too, but what's it used for? And can we make a change similar to graphql where we look up id columns from the entity type service and merge them into the hardcoded ones (since the id attribute is part of the metdata for entities that have been migrated to the v2 type service)

Reply:
The entity id is used for the execution tree in order to group by entity id for the aggregation requests. If we want to use entity type service, does that work with dynamic entity types like DOMAIN? Also introducing a new service dependency should be done in a new PR.

Reply:
Yep, agree it's a separate piece of work, and it doesn't replace the hardcoded config until all entities are migrated. Entity type service doesn't distinguish between projections, that's orthogonal.
Graphql, implements it as a dynamic loader: https://github.com/hypertrace/hypertrace-graphql/blob/main/hypertrace-graphql-entity-type/src/main/java/org/hypertrace/graphql/entity/EntityIdMappingLoader.java

Which gets combined with the static mappings: https://github.com/hypertrace/hypertrace-core-graphql/blob/3fc65784f6a37bcca8c069eb4e88eee9b063c772/hypertrace-core-graphql-attribute-store/src/main/java/org/hypertrace/core/graphql/attributes/IdLookup.java

Reply:
We used to have this in code as well till we changed to config so that we could support different entities dynamically. So do we proceed with this for now @aaron-steinfeld @buchi-busireddy ?

Reply:
sure

Reply:
Good discussion and my first point as well. Logged https://traceableai.atlassian.net/browse/ENG-7183 so that we don't miss it.